diff -u -r a/enc28j60.c b/enc28j60.c
--- a/enc28j60.c	2015-07-10 00:46:48.000000000 +0000
+++ b/enc28j60.c	2015-07-27 08:39:30.000000000 +0000
@@ -1581,7 +1581,8 @@
 	/* Board setup must set the relevant edge trigger type;
 	 * level triggers won't currently work.
 	 */
-	ret = request_irq(spi->irq, enc28j60_irq, 0, DRV_NAME, priv);
+	//ret = request_irq(spi->irq, enc28j60_irq, 0, DRV_NAME, priv);
+	ret = request_irq(spi->irq, enc28j60_irq, IRQF_TRIGGER_FALLING, DRV_NAME, priv);
 	if (ret < 0) {
 		if (netif_msg_probe(priv))
 			dev_err(&spi->dev, DRV_NAME ": request irq %d failed "
